### PR TITLE
chore: update contribution part about editor definition

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Add a part about **Adding a Che Editor to Che** into CONTRIBUTE.md file

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22854

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -31,109 +31,168 @@ Che supports multiple editors and each workspace can use its own editor. In the
...
+          endpoints:
+            # Name of the editor 
+            - name: che-code
+              # (OPTIONAL) Map of implementation-dependant string-based free-form attributes
+              attributes:
+                # Type of the endpoint
+                type: main
+                cookiesAuthEnabled: true
+                discoverable: false
+                urlRewriteSupported: true
```
Comment (svor) [inline CONTRIBUTE.md:151]: TODO
need to document these attributes

**Comment by svor** [comment]
@azatsarynnyy @tolusha I'd like to have your review as well to check that all attributes described properly

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -31,109 +31,173 @@ Che supports multiple editors and each workspace can use its own editor. In the
...
+        # Mandatory name that allows referencing the component from other elements
+      - name: checode
+        # (OPTIONAL) Allows specifying the definition of a volume shared by several other components. Ephemeral volumes are not stored persistently across restarts. Defaults to false
+        volume: {ephemeral: true}
+    # (OPTIONAL) Bindings of commands to events. Each command is referred-to by its name
+    events:
+      # IDs of commands that should be executed before the devworkspace start. This commands would typically be executed in init container
+      preStart:
+        - init-container-command
+      # IDs of commands that should be executed after the devworkspace is completely started. In the case of Che-Theia, these commands should be executed after all plugins and extensions have started, including project cloning. This means that those commands are not triggered until the user opens the IDE in his browser
```
Comment (azatsarynnyy) [inline CONTRIBUTE.md:174]: Should we mention Che-Code in this comment?
As Che-Theia isn't part of Che anymore and the example itself relates to Che-Code.

**Review by azatsarynnyy — APPROVED** [review]

**Pull request merged**